File: app/src/main/java/com/egecius/ui_playground/ImageUtils.java
package com.egecius.ui_playground;

import android.content.Context;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;
import android.support.v4.content.ContextCompat;
import android.support.v4.graphics.drawable.DrawableCompat;

public class ImageUtils {

    Drawable toTintedDrawable(Context context, @DrawableRes int drawableRes,
            @ColorRes int colorRes) {
        Drawable drawable = toDrawable(context, drawableRes);
        return toTintedDrawable(context, colorRes, drawable);
    }

    private Drawable toDrawable(Context context, @DrawableRes int drawableId) {
        return ContextCompat.getDrawable(context, drawableId);
    }

    private Drawable toTintedDrawable(Context context, @ColorRes int colorRes, Drawable mDrawable) {
        int colorInt = ContextCompat.getColor(context, colorRes);
        mDrawable.mutate();
        Drawable mWrappedDrawable = DrawableCompat.wrap(mDrawable);
        DrawableCompat.setTint(mWrappedDrawable, colorInt);
        DrawableCompat.setTintMode(mWrappedDrawable, PorterDuff.Mode.SRC_IN);
        return mWrappedDrawable;
    }
}

File: app/src/main/java/com/egecius/ui_playground/ImageUtils.kt
package com.egecius.ui_playground

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.annotation.ColorRes
import android.support.annotation.DrawableRes
import android.support.v4.content.ContextCompat
import android.support.v4.graphics.drawable.DrawableCompat

class ImageUtils {

    fun toTintedDrawable(context: Context, @DrawableRes drawableToTint: Int,
                         @ColorRes tintColor: Int): Drawable {
        val drawable = toDrawable(context, drawableToTint)
        return toTintedDrawable(context, tintColor, drawable)
    }

    private fun toDrawable(context: Context, @DrawableRes drawableId: Int): Drawable {
        return ContextCompat.getDrawable(context, drawableId)
    }

    private fun toTintedDrawable(context: Context, @ColorRes colorRes: Int, mDrawable: Drawable): Drawable {
        val colorInt = ContextCompat.getColor(context, colorRes)
        mDrawable.mutate()
        val mWrappedDrawable = DrawableCompat.wrap(mDrawable)
        DrawableCompat.setTint(mWrappedDrawable, colorInt)
        return mWrappedDrawable
    }
}
